### PR TITLE
bpo-1635741: Port _json extension module to multiphase initialization(PEP 489)

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue1635741

### CODE REVIEW

**Comment by vstinner** [comment]
cc @encukou @Dormouse759

File: Modules/_json.c [hunk 1]
```diff
@@ -1863,40 +1863,48 @@ static PyMethodDef speedups_methods[] = {
...
+    if (PyModule_AddObject(module, "make_encoder", (PyObject*)&PyEncoderType) < 0) {
         Py_DECREF((PyObject*)&PyEncoderType);
         goto fail;
     }
-    return m;
+    return 0;
   fail:
-    Py_DECREF(m);
-    return NULL;
+    Py_DECREF(module);
```
Comment (vstinner) [inline Modules/_json.c]: DECREF on error looks like a bug. For example, arraymodule.c doesn't do that. I suggest to remove the "fail" label and replace "goto fail" with "return -1".

File: Misc/NEWS.d/next/Core and Builtins/2020-01-05-13-40-08.bpo-1635741.QRTJVC.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Port _json extension module to multiphase initialization(PEP 489).
```
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2020-01-05-13-40-08.bpo-1635741.QRTJVC.rst]:
```suggestion
Port _json extension module to multiphase initialization (:pep:`489`).
```

File: Modules/_json.c (hunk 1 above)
Comment (shihai1991) [reply Modules/_json.c]: Good catch, thanks.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**